src/main/java/at/hannibal2/skyhanni/features/misc/TabWidgetSettings.kt

```diff
@@ -0,0 +1,63 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.utils.InventoryUtils.getInventoryName
+import at.hannibal2.skyhanni.utils.InventoryUtils.getUpperItems
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.inventory.ContainerChest
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class TabWidgetSettings {
+
+    private val mainPageWidgetPattern = "^§7Currently:.*".toRegex()
+
+    private val subPageWidgetPattern = "^§eClick to .*".toRegex(RegexOption.IGNORE_CASE)
+
+    private val shownSettingPattern = "Shown .* Setting.*".toRegex(RegexOption.IGNORE_CASE)
+
+    @SubscribeEvent
+    fun onBackgroundDrawn(event: GuiContainerEvent.BackgroundDrawnEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (event.gui !is GuiChest) return
+        if (!SkyHanniMod.feature.misc.highlightWidgets) return
+
+        val chest = event.gui.inventorySlots as ContainerChest
+        val inventoryName = chest.getInventoryName()
+        if (inventoryName.startsWith("Widgets ")) {
```

Comment:
Only use patterns. Every string we get from hypixel (direct or indriect) should be done as a RepoPattern, since it allows us to change the pattern remotely.